components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+					'd2l-calendar-date-selected': checkIfDatesEqual(day, this.selectedValue),
+					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
+				};
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				return html`
+					<td>
+						<div
+							@click="${this._onDateSelected}"
+							@keydown="${this._onKeyDown}"
+							class=${classMap(classes)}
+							data-date=${day.getDate()}
+							data-month=${day.getMonth()}
+							data-year=${day.getFullYear()}
+							tabindex=${focused ? '0' : '-1'}>
+							${day.getDate()}
+						</div>
+					</td>`;
+			});
+
+			return html`
+				<tr>${weekHtml}</tr>
+			`;
+		});
+		const heading = `${this._descriptor.calendar.months.long[this._shownMonth]} ${this._shownYear}`;
+		return html`
+			<div aria-labelledby="${this._dialogLabelId}" class="d2l-calendar">
+				<table summary="${this.summary}" role="grid">
+					<thead>
+						<tr class="d2l-calendar-title">
+							<td>
+								<d2l-button-icon
+									@click="${this._onPrevMonthButtonClick}"
+									text="${this._computeText(getPrevMonth(this._shownMonth))}"
+									icon="tier1:chevron-left">
+								</d2l-button-icon>
+							</td>
+							<th colspan="5">
+								<h2 class="d2l-heading-4" aria-live="polite" id="${this._dialogLabelId}">${heading}</h2>
+							</th>
+							<td>
+								<d2l-button-icon
+									@click="${this._onNextMonthButtonClick}"
+									text="${this._computeText(getNextMonth(this._shownMonth))}"
+									icon="tier1:chevron-right">
+								</d2l-button-icon>
+							</td>
+						</tr>
+					</thead>
+					<tbody>
+						<tr>
+							${weekdayHeaders}
+						</tr>
+						${dayRows}
+					</tbody>
+				</table>
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	_computeText(month) {
+		return this.localize('show', {month: this._descriptor.calendar.months.long[month]});
+	}
+
+	_focusDateAddFocus() {
+		const date = this._getFocusDateElement();
+		if (date) {
+			date.setAttribute('tabindex', '0');
+			if (this._keyboardTriggeredMonthChange) {
+				date.focus();
+				this._keyboardTriggeredMonthChange = false;
+			}
+		}
+	}
+
+	_focusDateRemoveFocus() {
+		const date = this._getFocusDateElement();
+		if (date) {
+			date.setAttribute('tabindex', '-1');
+		}
+	}
+
+	_getFocusDateElement() {
+		return this.shadowRoot.querySelector(`div[data-date="${this._focusDate.getDate()}"][data-month="${this._focusDate.getMonth()}"][data-year="${this._focusDate.getFullYear()}"]`);
+	}
+
+	_onDateSelected(e) {
+		this._focusDateRemoveFocus();
+		const selectedDate = e.composedPath()[0];
+		const year = selectedDate.getAttribute('data-year');
+		const month = selectedDate.getAttribute('data-month');
+		const date = selectedDate.getAttribute('data-date');
+		this.selectedValue = new Date(year, month, date);
+		this._focusDate = new Date(year, month, date);
+
+		const eventDetails = {
+			bubbles: true,
+			composed: true,
+			detail: { date: `${year}-${parseInt(month) + 1}-${date}` }
```

Comment:
Looks like the format here doesn't match the format in the readme for this event.  Also, should this date property be an actual `Date` or should we leave it as a `String`?

Reply:
As mentioned above, I think `selectedValue` should always be that same ISO string (and not be converted to a Date) at boot-up time. Then the consumer can always get the ISO value by accessing it. That would make the event details less useful, but we could still include the ISO value and the Date value in the details for simplicity.